base/src/dispatcher/dispatcher.ts

```diff
@@ -448,6 +463,15 @@ export class Dispatcher implements ZLUX.Dispatcher {
       return Promise.reject("no launch callback established");
     }
     let launchMetadata = this.buildObjectFromTemplate(action.primaryArgument, eventContext);
+    if (this.postMessageCallback && plugin.getWebContent().framework === 'iframe') {
+      console.log(`Is an iframe`);
+      let contexts = this.pendingIframes.get(plugin.getIdentifier());
+      if (!contexts) {
+        contexts = [];
+        this.pendingIframes.set(plugin.getIdentifier(), contexts);
```

Comment:
This needs an expiration so that we dont leak memory on things that never say they are ready.